python/pyiceberg/catalog/base_aws_catalog.py

```diff
@@ -0,0 +1,163 @@
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing,
+#  software distributed under the License is distributed on an
+#  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+#  KIND, either express or implied.  See the License for the
+#  specific language governing permissions and limitations
+#  under the License.
+
+import uuid
+from abc import ABC, abstractmethod
+from typing import (
+    Optional,
+    Set,
+    Tuple,
+    Union,
+)
+
+from pyiceberg.catalog import (
+    MANIFEST,
+    MANIFEST_LIST,
+    METADATA,
+    PREVIOUS_METADATA,
+    Catalog,
+    Identifier,
+    Properties,
+    PropertiesUpdateSummary,
+    delete_data_files,
+    delete_files,
+)
+from pyiceberg.io import FileIO, load_file_io
+from pyiceberg.partitioning import UNPARTITIONED_PARTITION_SPEC, PartitionSpec
+from pyiceberg.schema import Schema
+from pyiceberg.serializers import ToOutputFile
+from pyiceberg.table import Table
+from pyiceberg.table.metadata import TableMetadata, new_table_metadata
+from pyiceberg.table.sorting import UNSORTED_SORT_ORDER, SortOrder
+from pyiceberg.typedef import EMPTY_DICT
+
+EXTERNAL_TABLE_TYPE = "EXTERNAL_TABLE"
+
+
+class BaseAwsCatalog(Catalog, ABC):
```

Comment:
I am not sure if we need to extract this base class. I agree there are common logic shared across catalogs, but I don't think there is too much in common between Dynamo and Glue catalog implementations. 

I remember we have talked about having a base.py class when doing glue catalog implementation, but I think we ended up not doing that and just added util functions for shared functionalities. Maybe we should continue with that pattern.

Reply:
Removed `BaseAwsCatalog` class. I moved things around a bit to still remove duplicate code. There is some duplicate code left in `create_table()` but I think it's fine.